azurerm/data_source_data_lake_store.go

```diff
@@ -0,0 +1,66 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmDataLakeStoreAccount() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmDateLakeStoreAccountRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"tier": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  false,
+			},
+
+			"tags": tagsForDataSourceSchema(),
+		},
+	}
+}
+
+func dataSourceArmDateLakeStoreAccountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
```

Comment:
Can you add a warning message her that it wasn't found?
`log.Printf("[WARN] DataLakeStoreAccount '%s' was not found (resource group '%s').", name, resourcGroup)`